ext/serializer.c

```diff
@@ -1017,6 +1045,27 @@ static void dd_serialize_array_metrics_recursively(zend_array *target, zend_stri
     dd_serialize_array_recursively(target, str, value, true);
 }
 
+static void dd_serialize_array_meta_struct_recursively(zend_array *target, zend_string *str, zval *value) {
+    char *data;
+    size_t size;
+
+    mpack_writer_t writer;
+    mpack_writer_init_growable(&writer, &data, &size);
+    int result = msgpack_write_zval(&writer, value, 5);
+    mpack_writer_destroy(&writer);
+
+    if (size == 0 || result == 0) {
+        return;
+    }
+
+    zval serialised;
+    ZVAL_STRINGL(&serialised, data, size);
+
+    dd_serialize_array_recursively(target, str, &serialised, false);
```

Comment:
So, `serialised` is always a string, the only thing dd_serialize_array_recursively does ... is then adding it to the hashtable - it was confusing to read, so just simplify it:
```suggestion
    zend_hash_update(target, str, &serialised);
```